DashAI/front/src/components/ConfigurableObject/FormInputs.jsx

```diff
@@ -0,0 +1,377 @@
+import Subform from "./Subform";
+import { styled } from "@mui/material/styles";
+
+const Input = styled(TextField)(({ theme }) => ({
+  minWidth: "20vw",
+}));
+
+export function Integer({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        type="number"
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Integer.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Integer.defaultProps = {
+  error: undefined,
+};
+
+// This field is designed to handle non-integer numeric inputs
+export function Number({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Number.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Number.defaultProps = {
+  error: undefined,
+};
+
+export function Float({ name, value, onChange, description, error }) {
+  return (
+    <div key={name}>
+      <Input
+        variant="outlined"
+        label={name}
+        name={name}
+        value={value}
+        type="number"
+        onChange={onChange}
+        error={error}
+        helperText={error}
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Float.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Float.defaultProps = {
+  error: undefined,
+};
+
+export function Text({ name, value, onChange, error, description }) {
+  return (
+    <div key={name}>
+      <Input
+        name={name}
+        label={name}
+        defaultValue={value}
+        onKeyUp={onChange}
+        error={error}
+        helperText={error}
+        margin="dense"
+      />
+      {generateTooltip(description)}
+    </div>
+  );
+}
+Text.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  description: PropTypes.string.isRequired,
+  error: PropTypes.string,
+};
+Text.defaultProps = {
+  value: "",
+  error: undefined,
+};
+
+export function Select({ name, value, onChange, error, description, options }) {
```

Comment:
Move to `front/src/components/ConfigurableObject/Inputs/SelectInput.jsx`

```suggestion
export function SelectInput({ name, value, onChange, error, description, options }) {
```